### ISSUE TITLE
IPFS staging failures

### ISSUE THREAD

**Issue by franckc** [OriginProtocol/dshop#522]
Backend uploads content to staging IPFS:
```2020-08-28T17:21:12.62901776Z [DEBUG] dshop.offerProcessor: Creating offer on IPFS```
Hash is Qmf9zZEqrpep6RbVD4BcrckKX5rTdYjBWLXpC4DvWMYsT6

Backend tries to load the offer data a few seconds later and fails with a timeout 60 sec later:
```
2020-08-28T17:21:25.638729435Z [INFO]  dshop.utils.handleLog: Fetching offer data with hash Qmf9zZEqrpep6RbVD4BcrckKX5rTdYjBWLXpC4DvWMYsT6
...
2020-08-28T17:22:25.644961769Z [ERROR] dshop.queues.eventsProcessor: Log handler failed: Error: IPFS gateway timeout
```

I also tried to fetch the hash manually via my browser and it failed.

There were also retries from the queue to process the offer again and they all failed with a timeout.

**Comment by franckc** [comment]
The same thing happened with hash Qmeq4PzpzoPCsmdVwUi4K63NMgWf8U8uzikZTerZkPn7us,

**Comment by franckc** [comment]
Not super high priority but impairs our ability to test on staging.

CC @mikeshultz

**Comment by mikeshultz** [comment]
> Using IPFS gateway https://ipfs.staging.originprotocol.com for fetching offer data

Unless I'm mistaken, this is not the same node that's being used for pinning/deploy by the backend.  Cross-node fetches take time, and connections between nodes are ephimerial(which is why it starts working better after a while).  Either request from the gateway of the node/cluster being used for pinning, or expect indefinite failures until those p2p connections are made.

**Comment by franckc** [comment]
Yes good point I think that's what's happening...

It's confusing that the shops may not use the same IPFS gateway as the deployer backend...
I think the reason this can happen is when we change the IPFS node in the Network settings. That does not update the shops config.json. 

Hopefully we won't change the node in Networks config moving forward...?

But perhaps we should run a back job to update all the shops to point to the new IPFS cluster whenever we update the cluster pointed to by the Network config?

**Comment by mikeshultz** [comment]
AFAIK, there is no connection between the nodes in backend settings and in config.json.  I think config.json is based on an initial template that I don't know a lot about except that it makes my local testing hell.

**Comment by franckc** [comment]
Did a bit of digging and omg, we really need to bring sanity to our config code....

It turns our at shop creation we get the default shop config from [scripts/configs.js](https://github.com/OriginProtocol/dshop/blob/master/backend/scripts/configs.js) ! 
See shops creation code:
  - https://github.com/OriginProtocol/dshop/blob/master/backend/routes/shops.js#L35
  - https://github.com/OriginProtocol/dshop/blob/master/backend/routes/shops.js#L516

**Comment by franckc** [comment]
Actually I need to clarify with you @mikeshultz what auth we have in place in front of our IPFS cluster... I think my statement for option 2 above is incorrect. Will ping you on discord...

**Comment by mikeshultz** [comment]
> Our own IPFS cluster: same credentials problem as Pinata

False.  `/api/v0/add` is still conditionally open for users on the cluster. ` fs.ogn.app` is a drop-in replacement for `ipfs.originprotocol.com`.

**Comment by franckc** [comment]
Discussed with @mikeshultz - here is the plan:
 - For all offer data, we'll use the new IPFS cluster. There is no auth required for uploading json data via the gateway.
 - GC is enabled on that cluster (and it is probably healthier to keep it this way) so we'll need to pin the offer data. A possibility is for the back-end to do that when it gets called to record a payment during the checkout flow.

**Comment by franckc** [comment]
I'm renaming this issue to reflect better the work required.

**Comment by mikeshultz** [comment]
I remembered a key difference with the new cluster.  We shouldn't need to manually pin anything with the new cluster.  `@origin/ipfs-proxy` has been configured to use go-ipfs's ipfs-proxy which, in the case of a request to `/api/v0/add`, adds the file to all nodes in the cluster and pins it.  I just double-checked to confirm my memory as well.

So, we should just be able to switch to the new cluster without having to do anything but change the config.

A *nice to have* at some point would be to copy files+pins again from the old node just for redundancy and accessibility on the new gateway, but probably not super important/urgent.

**Comment by franckc** [comment]
Ah cool. Thanks for that details - makes things easier :)

And yes agree would be good to write a backfill script to move content from old node to new cluster. I have already captured this task above.
